web/b3desk/endpoints/captcha.py

```diff
@@ -0,0 +1,73 @@
+import requests
+from flask import Blueprint
+from flask import current_app
+from flask import jsonify
+from flask import request
+from flask import session
+
+bp = Blueprint("captcha", __name__)
+
+
+def get_captchetat_token():
+    url = f"{current_app.config['PISTE_OAUTH_API_URL']}/oauth/token"
+    form_data = {
+        "grant_type": "client_credentials",
+        "client_id": current_app.config["PISTE_OAUTH_CLIENT_ID"],
+        "client_secret": current_app.config["PISTE_OAUTH_CLIENT_SECRET"],
+        "scope": "piste.captchetat",
+    }
+    headers = {"Content-Type": "application/x-www-form-urlencoded"}
+    response = requests.post(url, data=form_data, headers=headers)
+    if response.status_code != 200 or "access_token" not in response.json():
+        session["visio_code"]["captcha_is_dead"] = True
+        current_app.logger.error("Captcha error: OAuth access token not received")
```

Comment:
In case of error, if available, please log the original error cause in the logs. It should be in `response.json()["error"]`.

Also, you should use a `try/except` to catch network level errors.